### PR TITLE
Feature/add regex column, addresses #125

### PR DESCRIPTION
**Pull request by lcorrigall**
For #125

### CODE REVIEW

File: src/rules/AddRegExColumn.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		}
+
+		if(!this.config.newColumn) {
+			this.error(`No 'newColumn' name defined.`);
+			return;
+		}
+
+		this.column = this.getValidatedColumnProperty();
+		this.badColumnCountReported = false;	// If a bad number of columns is found report it only once, not once per record.
+		this.reportAlways = this.config.reportAlways || true;	// Should every occurrence be reported?
```
Comment (ptokarchuk-uncharted) [inline src/rules/AddRegExColumn.js]: Won't this always evaluate to 'true'?

File: src/rules/AddRegExColumn.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	get processHeaderRows() {
+		return true;
+	}
+
+	start() {
+		// Because of the asynchronous nature of streams this modification of the shared data must be done
+		// before the rule starts rather than at the end. Otherwise following rules would start with the unmodified
+		// version of the shared data. Not what is desired.
+
+		// Remove the column label from the shared list of column labels.
```
Comment (ptokarchuk-uncharted) [inline src/rules/AddRegExColumn.js]: Change "Remove" to "Add".

File: src/rules/AddRegExColumn.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+		} else	if (this.column !== undefined) {
+			if (this.column >= record.length) {	// Does the record have the correct number of columns?
+				if (this.reportAlways || !this.badColumnCountReported) {
+					this.error(`Row ${rowId} has insufficient columns.`);
+					this.badColumnCountReported = true;
+				}
+			} else if(this.exec) {
+				const result = this.exec(record[this.column]);
+
+				if(this.onFailure && !result) {
```
Comment (ptokarchuk-uncharted) [inline src/rules/AddRegExColumn.js]: These next two 'if's can be combined to simplify things a little.

File: src/rules/CSVParser.js [hunk 4]
```diff
@@ -97,11 +97,12 @@ class CSVParser extends TableParserAPI {
             }
 
             let response = record;
+            let isHeaderRow = rowNumber < rowHeaderOffset;
 
             this.tableRule.resetLastCheckCounts();
 
-            if (this.tableRule && rowNumber >= rowHeaderOffset || processHeaderRows) {
-                response = this.tableRule.processRecordWrapper(record, rowNumber);
+            if (this.tableRule && (!isHeaderRow || processHeaderRows)) {
```
Comment (ptokarchuk-uncharted) [inline src/rules/CSVParser.js]: Line 102 assumes this.tableRule exists. So either the test in the 'if' is unnecessary or there's a potential crash in 102.

File: src/rules/AddRegExColumn.js (hunk 1 above)
Comment (lcorrigall) [reply src/rules/AddRegExColumn.js]: yes, removed

**Pull request merged**